splunk-otel-android/src/main/java/com/splunk/rum/SpanFilterBuilder.java

```diff
@@ -0,0 +1,130 @@
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+
+/**
+ * Allows to modify span data before it is sent to the exported. Spans can be modified or entirely
+ * rejected from export.
+ */
+public final class SpanFilterBuilder {
+
+    private Predicate<String> rejectSpanNamesPredicate = spanName -> false;
+    private final Map<AttributeKey<String>, Predicate<String>> rejectSpanAttributesPredicates = new HashMap<>();
+    private final Map<AttributeKey<String>, Function<String, String>> spanAttributeReplacements = new HashMap<>();
+
+    SpanFilterBuilder() {
+    }
+
+    /**
+     * Remove matching spans from the exporter pipeline.
+     * <p>
+     * Span names that match the {@code spanNamePredicate} will not be exported.
+     *
+     * @param spanNamePredicate A function that returns true if a span with passed name should be
+     *                          rejected.
+     * @return {@code this}.
+     */
+    public SpanFilterBuilder rejectSpansByName(Predicate<String> spanNamePredicate) {
+        rejectSpanNamesPredicate = rejectSpanNamesPredicate.or(spanNamePredicate);
+        return this;
+    }
+
+    /**
+     * Remove matching spans from the exporter pipeline.
+     * <p>
+     * Span that contains an attribute with key {@code attributeKey} and value matching the
+     * {@code attributeValuePredicate} will not be exported.
+     *
+     * @param attributeKey            An attribute key to match.
+     * @param attributeValuePredicate A function that returns true if a span containing an attribute
+     *                                with matching value should be rejected.
+     * @return {@code this}.
+     */
+    public SpanFilterBuilder rejectSpansByAttributeValue(
+            AttributeKey<String> attributeKey,
+            Predicate<String> attributeValuePredicate) {
+
+        rejectSpanAttributesPredicates.compute(attributeKey,
+                (k, oldValue) -> oldValue == null
+                        ? attributeValuePredicate
+                        : oldValue.or(attributeValuePredicate));
+        return this;
+    }
+
+    public SpanFilterBuilder removeSpanAttribute(AttributeKey<String> attributeKey) {
+        return removeSpanAttribute(attributeKey, value -> true);
+    }
+
+    /**
+     * Modify span data before it enters the exporter pipeline.
+     * <p>
+     * An attribute with key {@code attributeKey} and value matching the
+     * {@code attributeValuePredicate} will be removed from the span before it is exported.
+     *
+     * @param attributeKey            An attribute key to match.
+     * @param attributeValuePredicate A function that returns true if an attribute with matching
+     *                                value should be removed from the span.
+     * @return {@code this}.
+     */
+    public SpanFilterBuilder removeSpanAttribute(
+            AttributeKey<String> attributeKey,
+            Predicate<String> attributeValuePredicate) {
```

Comment:
could we make this use a parameterized type, rather than forcing `<String>` ?

Reply:
Yeah, parameterized types work, changed in all methods.
(Java generics are super painful to deal with when you have to use `?` wildcards though...)

Reply:
agreed. it's definitely not easy to figure out.